python/cudf/cudf/core/buffer/buffer.py

```diff
@@ -117,11 +117,31 @@ class BufferOwner(Serializable):
     # The set of buffers that point to this owner.
     _slices: weakref.WeakSet[Buffer]
 
-    def __init__(self):
-        raise ValueError(
-            f"do not create a {self.__class__} directly, please "
-            "use the factory function `cudf.core.buffer.as_buffer`"
-        )
+    def __init__(
+        self,
+        ptr: int,
+        size: int,
+        owner: object,
+        exposed: bool,
+    ):
+        """Create a new buffer owner.
+
+        Do not use this directly, instead use `_from_device_memory` or
+        `_from_host_memory`.
+
+        Raises
+        ------
+        ValueError
+            If size is negative
+        """
+        if size < 0:
+            raise ValueError("size cannot be negative")
+
+        self._ptr = ptr
+        self._size = size
+        self._owner = owner
+        self._exposed = exposed
+        self._slices = weakref.WeakSet()
```

Comment:
Notice, I have removed the direct-use-of-Buffer guard. I think the four mandatory arguments are enough to prevent accidental use.